### PR TITLE
Modify to limit user setting and values that may cause mismatching

### PR DESCRIPTION
**Pull request by HunsupJung**
Check all that apply

# Type of Change

- [ ] WWST Certification Request
     - If this is your first time contributing code:
          - [ ] I have reviewed the README.md file
          - [ ] I have reviewed the CODE_OF_CONDUCT.md file
          - [ ] I have signed the CLA
     - [ ] I plan on entering a WWST Certification Request or have entered a request through the WWST Certification console at developer.smartthings.com
- [x] Bug fix
- [ ] New feature
- [ ] Refactor

# Checklist

- [ ] I have performed a self-review of my code
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have verified my changes by testing with a device or have communicated a plan for testing
- [ ] I am adding new behavior, such as adding a sub-driver, and have added and run new unit tests to cover the new behavior

# Description of Change
The max limit of heating and cooling setpoint should be set to 39 or less because the value of 40 or more in the driver is considered Fahrenheit. Conversely, since a value of 40 or less is regarded as Celsius, it should not be possible to set the value below 40 degrees in the plugin in the Fahrenheit state. In summary, it should not be set above 40 degrees Celsius to below 5 degrees Celsius(40 degrees Fahrenheit).

# Summary of Completed Tests

# Ticket
[IOTE-4490](https://smartthings.atlassian.net/browse/IOTE-4490)

[IOTE-4490]: https://smartthings.atlassian.net/browse/IOTE-4490?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by HunsupJung** [comment]
Please check the following ticket
https://smartthings.atlassian.net/browse/IOTE-4490

**Comment by nickolas-deboom** [comment]
Hi @HunsupJung , I agree with these changes, however I have a PR [here](https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/1619) that implements dynamic constraints for matter-thermostat and matter-sensor that does many of the same things as your PR (in terms of the new handlers for the heating and cooling setpoint limits). Would you mind waiting until that PR is merged (should be by tomorrow) and then applying your changes for these min and max values afterwards?

**Comment by nickolas-deboom** [comment]
> Hi @HunsupJung , I agree with these changes, however I have a PR [here](https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/1619) that implements dynamic constraints for matter-thermostat and matter-sensor that does many of the same things as your PR (in terms of the new handlers for the heating and cooling setpoint limits). Would you mind waiting until that PR is merged (should be by tomorrow) and then applying your changes for these min and max values afterwards?

PR 1619 was merged into main and so now if you rebase you will be able to apply the changes for the temp limits on top of that.

**Comment by HunsupJung** [comment]
@nickolas-deboom 
Thank you for your explanation. I updated this PR

File: drivers/SmartThings/matter-thermostat/src/init.lua [hunk 1]
```diff
@@ -1130,16 +1136,25 @@ local heating_setpoint_limit_handler_factory = function(minOrMax)
...
+    end
+    if max > DEFAULT_MAX_SETPOINT then
+      max= DEFAULT_MAX_SETPOINT
+    end
+    if min < max then
+      -- Only emit the capability for RPC version >= 5 (unit conversion for
+      -- heating setpoint range capability is only supported for RPC >= 5)
+      if version.rpc >= 5 then
+        device:emit_event_for_endpoint(ib.endpoint_id, capabilities.thermostatHeatingSetpoint.heatingSetpointRange({ value = { minimum = min, maximum = max }, unit = "C" }))
+        device:set_field(setpoint_limit_device_field.MIN_HEAT, min)
```
Comment (nickolas-deboom) [inline drivers/SmartThings/matter-thermostat/src/init.lua:1153]: Should these `set_field` statements be placed outside of the if statement? Even if we don't set the capability we could still set the min and max values to their limited values.

File: drivers/SmartThings/matter-thermostat/src/init.lua [hunk 2]
```diff
@@ -1153,16 +1168,25 @@ local cooling_setpoint_limit_handler_factory = function(minOrMax)
...
+    end
+    if max > DEFAULT_MAX_SETPOINT then
+      max= DEFAULT_MAX_SETPOINT
+    end
+    if min < max then
+      -- Only emit the capability for RPC version >= 5 (unit conversion for
+      -- cooling setpoint range capability is only supported for RPC >= 5)
+      if version.rpc >= 5 then
+        device:emit_event_for_endpoint(ib.endpoint_id, capabilities.thermostatCoolingSetpoint.coolingSetpointRange({ value = { minimum = min, maximum = max }, unit = "C" }))
+        device:set_field(setpoint_limit_device_field.MIN_HEAT, min)
```
Comment (nickolas-deboom) [inline drivers/SmartThings/matter-thermostat/src/init.lua:1185]: Same thing here

**Comment by nickolas-deboom** [comment]
I left one comment but other than that the changes look great! Thanks for making the update.

**Comment by HunsupJung** [comment]
@nickolas-deboom 
this PR can be closed by [your PR](https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/1678). Thank you for your effort. 👍

**Pull request closed without merging**